src/io/cpal.rs

```diff
@@ -66,6 +65,35 @@ mod private {
 }
 use private::ThreadSafeClosableStream;
 
+fn get_device_label(sink_id: &str) -> String {
+    if !sink_id.is_empty() {
+        let device = CpalBackend::enumerate_devices_sync()
+            .into_iter()
+            .find(|e| e.device_id() == sink_id)
+            .unwrap();
+
+        return device.label().to_string();
+    }
+
+    "default".to_string()
+}
+
+// label is only used when the cpal-jack feature is activated
+#[allow(unused_variables)]
+fn get_host(label: &str) -> cpal::Host {
```

Comment:
You should probably call `get_host` as well in `enumerate_devices_sync` down below no, instead of `cpal::default_host`?
Otherwise you ask Alsa (when it is default host) for the devices, and then construct them via Jack, no?

And I'm not too sure why get_host should depend on the label.
Why not choose jack when it is available, and default_host otherwise?

Like so:
```rust
fn get_host() -> cpal::Host {
    #[cfg(feature = "cpal-jack")]
    {
        if let Some(jack_available) = cpal::available_hosts()
            .into_iter()
            .find(|id| *id == cpal::HostId::Jack) {
                 return cpal::host_from_id(jack_available).unwrap();
            };
    }
    cpal::default_host()
}

In that case `get_device_label` is not necessary at all.

Reply:
Yup you are right, I didn't try to see what `enumerate_device_sync` using jack host would give us actually, I will check that. Maybe it can simplify the setup yes (I must confess the current version is mostly "try this, try that, works, don't touch!" :)